source/docs/software/basic-programming/joystick.rst

```diff
@@ -0,0 +1,151 @@
+.. include:: <isonum.txt>
+
+Joysticks
+=========
+
+In FRC\ |reg| almost any "controller" that can be recognized by Windows can be used as a joystick.  Joysticks are accessed using the ``GenericHID`` class.  This class has two relevant subclasses for preconfigured joysticks but if not listed you could implement your own.  The first is ``Joystick`` which is useful for standard flight joysticks.  The second is ``XboxController`` which works for the Xbox 360, Xbox One, or Logitech F310 (in XInput mode).  Each axis of the controller ranges from -1 to 1.
```

Comment:
```suggestion
In FRC\ |reg|, almost any "controller" that can be recognized by Windows can be used as a joystick.  Joysticks are accessed using the ``GenericHID`` class.  This class has two relevant subclasses for preconfigured joysticks but if not listed you could implement your own.  The first is ``Joystick`` which is useful for standard flight joysticks.  The second is ``XboxController`` which works for the Xbox 360, Xbox One, or Logitech F310 (in XInput mode).  Each axis of the controller ranges from -1 to 1.
```

Reply:
I would mention that Joystick class is useful beyond just flight controllers. It's basically essential for any non-xbox controller

Reply:
I completely disagree with this as I currently understand things.  It is useful for every type of joystick but beyond that not really.

Reply:
No. You have to use the Joystick class for Playstation controllers, switch controllers, etc. You cannot use Xbox controller as the bindings will be incorrect